easypermissions/src/main/java/pub/devrel/easypermissions/RationaleAppCompatDialogFragment.java

```diff
@@ -0,0 +1,122 @@
+package pub.devrel.easypermissions;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AlertDialog;
+
+import java.util.Arrays;
+
+
+/**
+ * RationaleAppCompatDialogFragment that is being used where the calling Activity/Fragment is from the support Library
+ */
+public class RationaleAppCompatDialogFragment extends DialogFragment implements Dialog.OnClickListener {
+
+    private EasyPermissions.PermissionCallbacks permissionCallbacks;
+
+    private int positiveButton;
+    private int negativeButton;
+    private int requestCode;
+    private String rationaleMsg;
+    private String[] permissions;
+
+    static RationaleAppCompatDialogFragment newInstance(@StringRes int positiveButton,
+                                                        @StringRes int negativeButton,
+                                                        @NonNull String rationaleMsg,
+                                                        int requestCode,
+                                                        @NonNull String[] permissions) {
+        RationaleAppCompatDialogFragment dialogFragment = new RationaleAppCompatDialogFragment();
+        Bundle bundle = new Bundle();
+        bundle.putInt("positiveButton", positiveButton);
```

Comment:
positiveButton & negativeButton are parameters are passed through the EasyPermissions constructer, you are given the user the option to initialise them, how we would make them final if they can be initialised.  could you please give me an example on how this could be ?